File: Online-judge/leetcode/find-anagram-mappings.py
from collections import defaultdict
from typing import List

class Solution:
    def anagramMappings(self, nums1: List[int], nums2: List[int]) -> List[int]:
        mapping = defaultdict(int)
        for index, num in enumerate(nums2):
            mapping[num] = index
        ans: List[int] = []
        for num in nums1:
            ans.append(mapping[num])
        return ans

File: Online-judge/leetcode/find-anagram-mappings.cpp
#include <bits/stdc++.h>
using namespace std;

/**
 * @author Md.Shohag <shohag.fks@gmail.com>
 * Problem : https://leetcode.com/problems/find-anagram-mappings/description/
 */

/**
 * @brief 
 * Time complexity : O(n)
 * Space complexity: O(n)
 */

class Solution {
public:
    vector<int> anagramMappings(vector<int>& nums1, vector<int>& nums2) {
        map<int, int> mapping;
        for (int index = 0; index < nums2.size(); index++) {
            mapping[nums2[index]] = index;
        }

        vector<int> ans;
        for (int num : nums1) {
            ans.push_back(mapping[num]);
        }
        return ans;
    }
};